### PR TITLE
Migrate to Python-based kavm runner script

### LINKED ISSUE #39: Replace `kavm` script with python runner script where it makes sense
Blocked on: https://github.com/runtimeverification/avm-semantics/issues/37

In order to support programmatic interaction with KAVM, we should adjust the `kavm` script to use the `kavm_pyk` stubs where possible. Good choices for this are [`kavm kompile ...`](https://github.com/runtimeverification/avm-semantics/blob/master/kavm#L45) , [`kavm run ...`](https://github.com/runtimeverification/avm-semantics/blob/master/kavm#L112),and [`kavm prove ...`](https://github.com/runtimeverification/avm-semantics/blob/master/kavm#L133). We should still have the bash script set up any environment nonsense it needs to before calling the python implementations of these functions.

This will:

1. Battle-test our `pyk.ktool` library.
2. Make it easier to port the summarizer to this codebase.
3. Make it easier to integrate with the new grant application to implement property testing: https://github.com/runtimeverification/avm-semantics/pull/36

### PR DESCRIPTION
**Pull request by geo2a**
* The `kavm_algod` Python package is renamed to `kavm` and now includes a `kavm` entry-point
* The bash-based `kavm` runner is renamed to `kavm.sh` and deprecated
* The Makefile and Jenkins now use the Python-based runner to call `kompile`, `run` and `kast`

Fixes #39

### CODE REVIEW

File: kavm/src/kavm/__main__.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    args = parser.parse_args()
+
+    if not args.definition_dir:
+        env_definition_dir = os.environ.get('KAVM_DEFINITION_DIR')
+        if env_definition_dir:
+            args.definition_dir = Path(env_definition_dir)
+        else:
+            raise RuntimeError('Cannot find KAVM definition, plese specify either --definition or KAVM_DEFINITION_DIR')
+
+    if args.command == 'kompile':
```
Comment (tothtamas28) [inline kavm/src/kavm/__main__.py:24]: Consider extracting a function for each of these commands. (You can check `kevm-pyk.__main__` for an [example](https://github.com/runtimeverification/evm-semantics/blob/e7a58e8e83883a6667ac0d798bd069b0321f16b4/kevm-pyk/src/kevm_pyk/__main__.py#L48).)

File: kavm/Makefile [hunk 2]
```diff
@@ -4,7 +4,7 @@
         format isort autoflake black             \
         check check-isort check-autoflake check-black check-flake8 check-mypy
 
-default: check test-unit
+default: build check test-unit
```
Comment (tothtamas28) [inline kavm/Makefile:7]:
```suggestion
default: check test-unit build
```

File: Makefile [hunk 3]
```diff
@@ -51,7 +53,7 @@ export K_OPTS
 .PHONY: all clean distclean install uninstall                                         \
         deps k-deps libsecp256k1 libff plugin-deps hook-deps                          \
         build build-avm build-kavm                                                    \
-        test test-avm                                                                 \
+        test test-avm-semantics                                                       \
         venv venv-clean kavm-algod
```
Comment (tothtamas28) [inline Makefile:57]: The list of `.PHONY` targets is not up to date, consider e.g. `kavm-algod`.

File: README.md [hunk 4]
```diff
@@ -1,36 +1,38 @@
-Algorand Virtual Machine and TEAL Semantics in K
+KAVM &mdash; Algorand Virtual Machine and TEAL Semantics in K
 ================================================
```
Comment (ehildenb) [inline README.md:2]: What's going on here? Either `===` need to be extended, or `&mdash;` removed.

File: Makefile [hunk 5]
```diff
@@ -50,12 +52,16 @@ export K_OPTS
...
         deps k-deps libsecp256k1 libff plugin-deps hook-deps                          \
-        build build-avm build-kavm                                                    \
-        test test-avm                                                                 \
-        venv venv-clean kavm-algod
+        build build-avm build-kavm py-kavm                                            \
+        test test-avm-semantics test-avm-semantics-prove                              \
+	test-kavm test-kavm-kast test-kavm-kast-avm-scenario test-kavm-kast-teal      \
+	clean-avm clean-kavm                                                          \
+	module-imports-graph module-imports-graph-dot                                 \
+        venv venv-clean
```
Comment (ehildenb) [inline Makefile:60]: Make sure your formatting the same as the surrounding code, and that you do self-review in GitHub before asking for review, to catch these space/tab issues. Use spaces here.

Comment (geo2a) [reply Makefile:60]: Sorry about that!

File: kavm/src/kavm/__main__.py [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+_LOG_FORMAT: Final = '%(levelname)s %(asctime)s %(name)s - %(message)s'
+
+
+def exec_kompile(
+    definition_dir: Path,
+    main_file: Path,
+    main_module: Optional[str],
+    syntax_module: Optional[str],
+    # includes: List[str] = [f'{install_include}/kframework', f'{plugin_include}/kframework'],
+    includes: List[str],
```
Comment (tothtamas28) [inline kavm/src/kavm/__main__.py:25]:
```suggestion
    includes: List[str],
```

File: kavm/src/kavm/kavm.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+class KAVM(KRun):
+    """
+    Interact with the K semantics of AVM: evaluate Algorand transaction groups
+    """
+
+    def __init__(
+        self,
+        definition_dir: Path,
+        use_directory: Any = None,
+        logger: Optional[logging.Logger] = None,
```
Comment (tothtamas28) [inline kavm/src/kavm/kavm.py:26]: Consider importing `Logger`.

```suggestion
        logger: Optional[Logger] = None,
```

File: kavm/src/kavm/__main__.py [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+    kompile_subparser.add_argument('main_file', type=Path, help='Path to the main file')
+    kompile_subparser.add_argument('--main-module', type=str)
+    kompile_subparser.add_argument('--syntax-module', type=str)
+
+    # kast
+    kast_subparser = command_parser.add_parser('kast', help='Kast a term', parents=[shared_args])
+    kast_subparser.add_argument(
+        '--definition',
+        dest='definition_dir',
+        type=Path,
```
Comment (tothtamas28) [inline kavm/src/kavm/__main__.py:153]: Module `pyk.cli` contains functions `file_path` and `dir_path` that check if the corresponding path is an existing file or directory, respectively:

```suggestion
        type=dir_path,
```

**Review by tothtamas28 — APPROVED** [review]
Looks good, left a few suggestions to consider.

**Pull request merged**